game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -1202,6 +1162,86 @@ && canAirSafelyLandAfterAttack(unit, t)) {
     }
   }
 
+  private void determineTerritoriesThatCanBeBombed(
+      final Map<Territory, ProTerritory> attackMap,
+      final Map<Unit, Set<Territory>> sortedUnitAttackOptions,
+      final Set<Unit> alreadyAttackedWithUnits) {
+    final Map<Unit, Set<Territory>> bomberMoveMap =
+        territoryManager.getAttackOptions().getBomberMoveMap();
+    final Set<Unit> bombers = bomberMoveMap.keySet();
+    bombers.removeAll(alreadyAttackedWithUnits);
+    final boolean raidsMayBePrecededByAirBattles =
+        Properties.getRaidsMayBePreceededByAirBattles(data.getProperties());
+    for (final Unit bomber : bombers) {
+      final Set<Territory> bomberTargetTerritories = bomberMoveMap.get(bomber);
+      bomberTargetTerritories.removeIf(
+          terr ->
+              raidsMayBePrecededByAirBattles
```

Comment:
Since this won't remove anything if `raidsMayBePrecededByAirBattles` is false, you can rework this to avoid the work if the property isn't set:

```
if (raidsMayBePrecededByAirBattles) {
  bomberTargetTerritories.removeIf(
          terr -> AirBattle.territoryCouldPossiblyHaveAirBattleDefenders(
                      terr, player, data, true));
}
```

Reply:
Additionally, this will modify the underlying map. I think you want to do `getMatches()` instead.

Reply:
Extracted `raidsMayBePrecededByAirBattles` as described.
Whole section reworked also taking into account the other comment.